File: AspNetMvcUnleashed/0672329980_source/Chapter02/CS/ToyStore/ToyStore/Controllers/HomeController.cs
using System.Linq;
using System.Web.Mvc;
using ToyStore.Models;

namespace ToyStore.Controllers
{
    public class HomeController : Controller
    {
        private ToyStoreDBEntities _dataModel = new ToyStoreDBEntities(); 
               
        //
        // GET: /Home/

        public ActionResult Index()
        {
            return View(_dataModel.ProductSet.ToList());
        }

        //
        // GET: /Home/Create

        public ActionResult Create()
        {
            return View();
        } 

        //
        // POST: /Home/Create

        [AcceptVerbs(HttpVerbs.Post)]
        public ActionResult Create([Bind(Exclude="Id")]Product productToCreate)
        {
            if (!ModelState.IsValid)
                return View();

            try
            {
                _dataModel.AddToProductSet(productToCreate);
                _dataModel.SaveChanges();

                return RedirectToAction("Index");
            }
            catch
            {
                return View();
            }
        }


    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter02/VB/ToyStore/ToyStore/Controllers/HomeController.vb
Public Class HomeController
    Inherits System.Web.Mvc.Controller

    Private _dataModel As New ToyStoreDBEntities()


    '
    ' GET: /Home/

    Function Index() As ActionResult
        Return View(_dataModel.ProductSet.ToList())
    End Function


    '
    ' GET: /Home/Create

    Function Create() As ActionResult
        Return View()
    End Function

    '
    ' POST: /Home/Create

    <AcceptVerbs(HttpVerbs.Post)> _
    Function Create(<Bind(Exclude:="Id")> ByVal productToCreate As Product) As ActionResult

        If Not ModelState.IsValid Then
            Return View()
        End If

        Try
            _dataModel.AddToProductSet(productToCreate)
            _dataModel.SaveChanges()

            Return RedirectToAction("Index")
        Catch
            Return View()
        End Try
    End Function


End Class
